jbpm-console-ng-human-tasks/jbpm-console-ng-human-tasks-client/src/main/java/org/jbpm/console/ng/ht/client/editors/taskslist/grid/dash/DataSetTasksListGridPresenter.java

```diff
@@ -162,70 +177,184 @@ public void getData( final Range visibleRange ) {
-                            GWT.log( "DataSet with UUID [  jbpmHumanTasks ] not found." );
+            @Override
+            public boolean onError( final ClientRuntimeError error ) {
+                view.hideBusyIndicator();
+                error.getThrowable().printStackTrace();
+                errorPopup.showMessage( "DataSet with UUID [  jbpmHumanTasks ] error: " + error.getThrowable() );
+                GWT.log( "DataSet with UUID [  jbpmHumanTasks ] error: ", error.getThrowable() );
+                return false;
+            }
+
+            @Override
+            public void callback(DataSet dataSet) {
+
+                if (dataSet != null) {
+                    final List<TaskSummary> myTasksFromDataSet = new ArrayList<TaskSummary>();
+
+                    for (int i = 0; i < dataSet.getRowCount(); i++) {
+                        myTasksFromDataSet.add(createTaskSummaryFromDataSet(dataSet, i));
+
+                    }
+                    List<DataSetOp> ops= tableSettings.getDataSetLookup().getOperationList();
+                    String filterValue = isFilteredByTaskName(ops); //Add here the check to add the domain data columns taskName?
+
+
+                    if (filterValue != null) {
+                        getDomainSpecifDataForTasks(startRange, dataSet.getRowCountNonTrimmed(),filterValue, myTasksFromDataSet);
+                    } else {
+                        PageResponse<TaskSummary> taskSummaryPageResponse = new PageResponse<TaskSummary>();
+                        taskSummaryPageResponse.setPageRowList(myTasksFromDataSet);
+                        taskSummaryPageResponse.setStartRowIndex(startRange);
+                        taskSummaryPageResponse.setTotalRowSize(dataSet.getRowCountNonTrimmed());
+                        taskSummaryPageResponse.setTotalRowSizeExact(true);
+                        if (startRange + dataSet.getRowCount() == dataSet.getRowCountNonTrimmed()) {
+                            taskSummaryPageResponse.setLastPage(true);
+                        } else {
+                            taskSummaryPageResponse.setLastPage(false);
                         }
 
-                        @Override
-                        public boolean onError( final ClientRuntimeError error ) {
-                            view.hideBusyIndicator();
-                            error.getThrowable().printStackTrace();
-                            errorPopup.showMessage( "DataSet with UUID [  jbpmHumanTasks ] error: " + error.getThrowable() );
-                            GWT.log( "DataSet with UUID [  jbpmHumanTasks ] error: ", error.getThrowable() );
-                            return false;
+                        updateDataOnCallback(taskSummaryPageResponse);
+                    }
+
+                }
+                view.hideBusyIndicator();
+
+            }
+
+
+
+        };
+    }
+
+    protected String isFilteredByTaskName(List<DataSetOp> ops) {
```

Comment:
I think this method should just return a boolean instead. Doesn' t seem like the actual String is relevant.